spec/models/credential_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Credential, type: :model do
+  let(:establishment) { create :establishment }
+  let(:login) { Faker::Internet.email }
+  let(:password) { Faker::Alphanumeric.alpha(number: 10) }
+  it 'is valid with attributes' do
+    credential = Credential.create(login: login,
+                                   password: password)
+    expect(credential).to be_valid
+  end
+
+  it 'is not valid without login' do
+    credential = Credential.create(password: password)
+    expect(credential).to_not be_valid
+  end
+
+  it 'is not valid without password' do
+    credential = Credential.create(login: login)
+    expect(credential).to_not be_valid
+  end
+
+  it 'is not valid wit same login' do
```

Comment:
You can structure this inside a new context where a credential with the same login exists
```
context 'when a credential already exists' do
  before do
    create(:credential, ...)
  end
  it 'is not valid with the same login' do
    ...
  end
end```